### ISSUE TITLE
[vim9script] duplicate name not always detected

### ISSUE THREAD

**Issue by errael** [vim/vim#13204]
### Steps to reproduce

The following script has `static sval` and `this.sval`. 
- The duplicate name is **not** detected when **processing the class**.
- The duplicate name is **not** detected when **running the static method**.
- The duplicate is finally detected when `C.new()` is executed.
- The duplicate is **not** detected when `C.new()` is manually defined and executed.

Source this script and notice that `C.F2()` works ok. The duplicate name is not detected.
```
vim9script

class C
    public static sval: list<number>
    public this.sval: list<number>
    def F1()
        echo this.sval
    enddef
    static def F2()
        echo sval
    enddef
endclass

C.F2()              # no error
var o = C.new()     # error
```

This gets an error
```
vim9script

class C
    public static sval: list<number>
    public this.sval: list<number>
endclass

var o = C.new()   # error
```
This gets no error
```
vim9script

class C
    public static sval: list<number>
    public this.sval: list<number>
    def new()
    enddef
endclass

var o = C.new()     # no error
```

Notice that in this example the duplicate method name is detected during processing the class
```
vim9script

class C
    def F1()
    enddef
    def F1()
    enddef
endclass
```

### Expected behaviour

Consistent detection of duplicate names in a class.

Consistent error messages would be nice. 

1. `Duplicate`
2. `Argument already declared`

The 2nd is confusing as well. `Argument`? (I get the automatically defined constructor)

### Version of Vim

9.0.1946

### Environment

ubuntu

**Comment by errael** [comment]
This is particularly weird, along the same vein. About declaring a variable named `this`.

This gets an argument shadows error
```
vim9script

class C
    public this.this: list<number> = [1, 2, 3]
endclass
var o = C.new()
```
This compiles no error
```
vim9script

class C
    public this.this: list<number> = [1, 2, 3]
    def new()
    enddef
endclass
var o = C.new()
```
This one outputs the list
```
vim9script

class C
    public this.this: list<number> = [1, 2, 3]
    def new()
    enddef
    def F()
        echo this.this
    enddef
endclass
var o = C.new()
o.F()
```

**Comment by errael** [comment]
All the errors in this issue fixed by #13209
